server/controllers/profile.js

```diff
@@ -0,0 +1,91 @@
+const ObjectId = require('mongoose').Types.ObjectId;
+const asyncHandler = require("express-async-handler");
+const Profile = require("../models/Profile");
+const User = require("../models/User");
+
+// @route POST /profile
+// @Given parameters passed in, create a profile.
+exports.createProfile = asyncHandler(async (req, res, next) => {
+  let profileData = req.body;
+  let profile = new Profile(profileData);
+
+  try {
+    const newProfile = await profile.save();
+    res.status(200).json({
+      success: {
+        profile: newProfile,
+      },
+    });
+  } catch (e) {
+    res.status(500);
+    throw new Error(e.message);
+  }
+});
```

Comment:
What do you think about just doing this no register instead of having a separate route?

Reply:
removed this part from profile controller